elasticdl/python/worker/task_data_service.py

```diff
@@ -83,9 +85,9 @@ def report_record_done(self, count, err_msg=""):
             if err_msg:
                 self._log_fail_records(task, err_msg)
 
-            # Keep poping: batch_size may be larger than
-            # task.end - task.start, so we keep poping task until
-            # reported count is less than current data size
+            # Keep popping tasks until the reported record count is less
```

Comment:
I think you want to mean "Keep popping tasks until the reported record count exceeds the size of the current data"?

Reply:
No that's not what I meant since the code is `self._reported_record_count >= (
                    self._pending_tasks[0].end - self._pending_tasks[0].start
                )`